pkg/operator/controller/callbacks.go

```diff
@@ -355,3 +360,47 @@ func restoreOlderVersions(currentCrd, desiredCrd *extv1.CustomResourceDefinition
 	}
 	return desiredCrd
 }
+
+func reconcilePvcMutatingWebhook(args *callbacks.ReconcileCallbackArgs) error {
+	if args.State != callbacks.ReconcileStatePostRead {
+		return nil
+	}
+
+	crd := args.CurrentObject.(*extv1.CustomResourceDefinition)
+	if crd.Name != "cdiconfigs.cdi.kubevirt.io" {
+		return nil
+	}
+
+	cdi, ok := args.Resource.(*cdiv1.CDI)
+	if !ok || cdi.Spec.Config == nil {
+		return nil
+	}
+
+	enabled := false
+	for _, fg := range cdi.Spec.Config.FeatureGates {
+		if fg == featuregates.WebhookPvcRendering {
+			enabled = true
+			break
+		}
+	}
+
+	whc := &admissionregistrationv1.MutatingWebhookConfiguration{}
+	crKey := client.ObjectKey{Name: "cdi-api-pvc-mutate"}
+	if err := args.Client.Get(context.TODO(), crKey, whc); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		}
+		if enabled {
+			whc = cluster.CreatePvcMutatingWebhook(args.Namespace, args.Client, args.Logger)
```

Comment:
I think to be consistent with the other resources created by callbacks, maybe put `CreatePvcMutatingWebhook` in this package.

Reply:
And don't forget to make sure it is owned by CDI

Reply:
Move it to this package although all other similar webhook creation code is in the cluster package? much hartder to maintain

Reply:
> And don't forget to make sure it is owned by CDI

`cluster-scoped resource must not have a namespace-scoped owner, owner's namespace cdi`

Reply:
> Move it to this package although all other similar webhook CR creation code is in the cluster package? much harder to maintain

Can you help me understand that?  To me, being in the same file or package is more maintainable for non-shared code

Reply:
Moved. I buy it when you explain it this way.